torch/_subclasses/fake_impls.py

```diff
@@ -329,6 +329,24 @@ def nonzero(fake_mode, func, arg):
 
     return arg.new_empty((arg.nonzero_memo, arg.dim()), dtype=torch.int64)
 
+@register_op_impl(torch.ops.aten._unique2.default)
+def unique(fake_mode, func, arg, **kwargs):
+    if (
+        fake_mode.shape_env is None
+        or not fake_mode.shape_env.allow_dynamic_output_shape_ops
+    ):
+        # Without symints/symfloats, cannot handle this
+        raise DynamicOutputShapeException(func)
+    
+    uni_cnt = fake_mode.shape_env.create_unbacked_symint()
+    # Avoid importing sympy at a module level
+    from torch.fx.experimental.symbolic_shapes import (
+        _constrain_range_for_size
+    )
+    _constrain_range_for_size(uni_cnt, min=1, max=arg.numel())
+    return arg.new_empty(uni_cnt), arg.new_empty(arg.size(), dtype=torch.int64), arg.new_empty(uni_cnt, dtype=torch.int64)
```

Comment:
The shape of the last two returned tensors depend on `return_inverse` and `return_counts`

Reply:
@eellison  thanks for your review!
`kwargs `doesn't carry `return_inverse` and `return_counts`, is this expected?
Also when I test with `torch.unique(x, sorted=True, return_inverse=True, return_counts=False)`, it throws errror "unique() takes 3 positional arguments but 5 were given". But `torch.unique(x, sorted=True, return_inverse=False, return_counts=False)` works good at the same time. do you happen to know why? Seems dispatch logic is having some problem to me.

Reply:
@eellison thanks for your review. The caller of _unique2 (File "/root/pytorch/torch/functional.py", line 910, in _unique_impl) requires three results regardless the value of return_inverse and return_counts
```
output, inverse_indices, counts = torch._unique2(
            input,
            sorted=sorted,
            return_inverse=return_inverse,
            return_counts=return_counts,
        )
```